packages/design-system/src/components/Dropdown/Dropdown.tsx

```diff
@@ -79,34 +95,47 @@ const Dropdown = forwardRef(
 
 						if (entry.type === 'title') {
 							const { label } = entry;
+							const id = `${label}-${index}`;
 							return (
-								<DropdownTitle key={`${label}-${index}`} data-test="dropdown.menuitem">
+								<DropdownTitle
+									key={id}
+									data-testid={`${menuItemTestId}.${id}`}
+									data-test={menuItemTestId}
+								>
 									{label}
 								</DropdownTitle>
 							);
 						}
 
 						if (entry.type === 'divider') {
+							const id = `divider-${index}`;
 							return (
-								<DropdownDivider {...menu} key={`divider-${index}`} data-test="dropdown.menuitem" />
+								<DropdownDivider
+									{...menu}
+									key={id}
+									data-testid={`${menuItemTestId}.${id}`}
+									data-test={menuItemTestId}
+								/>
 							);
 						}
 
 						const { label, as, type, ...entryRest } = entry;
+						const id = `${label}-${index}`;
 						return (
 							<DropdownLink
 								as={as}
 								{...entryRest}
 								{...menu}
-								key={`${label}-${index}`}
-								id={`${label}-${index}`}
+								key={id}
+								id={id}
 								onClick={(event: MouseEvent<HTMLAnchorElement>) => {
 									menu.hide();
 									if (entry.onClick) {
 										entry.onClick(event);
 									}
 								}}
-								data-test="dropdown.menuitem"
+								data-testid={`${menuItemTestId}.${id}`}
+								data-test={menuItemTestId}
```

Comment:
```suggestion
								data-test={`${menuItemTestId}.${id}`}
```

Reply:
this is supposed to be the same everywhere no ?

Reply:
don't know, changing the data-test could break current usage of it

Reply:
updated it as you said, as not a lot of usage for now